wheeltec:
  # 上面的 wheeltec 是运行命名空间
  senior_akm_joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Velocity Controllers ----速度控制器---------------------
  senior_akm_left_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: left_rear_wheel_joint

  senior_akm_right_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: right_rear_wheel_joint

  senior_akm_left_front_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: left_front_wheel_joint

  senior_akm_right_front_wheel_velocity_controller:
    type: velocity_controllers/JointVelocityController
    joint: right_front_wheel_joint

  # Position Controllers ---位置控制器-----------------------
  senior_akm_left_steering_hinge_position_controller:
    joint: left_steering_joint
    type: position_controllers/JointPositionController

  senior_akm_right_steering_hinge_position_controller:
    joint: right_steering_joint
    type: position_controllers/JointPositionController
#   gazebo_ros_control:
#     pid_gains:
#       left_steering_joint: { p: 1, i: 0.0, d: 0.0 }
#       right_steering_joint: { p: 1, i: 0.0, d: 0.0 }
#       left_rear_wheel_joint: { p: 1.5, i: 0.0, d: 0.0 }
#       right_rear_wheel_joint: { p: 1.5, i: 0.0, d: 0.0 }
